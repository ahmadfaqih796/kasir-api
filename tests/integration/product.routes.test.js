const request = require('supertest');
const app = require('../../src/app');

describe('Product API Endpoints', () => {
  test('GET /products should return product list', async () => {
    const res = await request(app).get('/products');
    expect(res.status).toBe(200);
    expect(Array.isArray(res.body.data)).toBe(true);
  });

  test('POST /products should create a product', async () => {
    const res = await request(app)
      .post('/products')
      .send({ name: 'Mouse', price: 20, stock: 50, created_by: 1 });
    expect(res.status).toBe(201);
    expect(res.body.name).toBe('Mouse');
    expect(Object.keys(res.body).length).toBe(7);
  });
});
