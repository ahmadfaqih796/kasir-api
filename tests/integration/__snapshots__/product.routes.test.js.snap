// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product API Endpoints POST /products should create a product 1`] = `
{
  "created_by": 1,
  "name": "Mouse",
  "price": 20,
  "stock": 50,
}
`;
